### PR TITLE
Add tidbmonitor remotewrite basic auth example

### PR DESCRIPTION
**Pull request by mikechengwei**
### What problem does this PR solve?

#4455 

### What is changed and how does it work?
Add tidbmonitor remotewrite basic auth example

### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

**Comment by mikechengwei** [comment]
Manual Test screenshot:
If i comment the basic auth configuration, the data write will throw 401 authorization error.

Screen Shot 2022-03-14 at 1 24 15 AM

Screen Shot 2022-03-14 at 1 24 02 AM

If i uncomment the the basic auth configuration, the data will write successfully.

Screen Shot 2022-03-14 at 1 25 08 AM
Screen Shot 2022-03-14 at 1 27 36 AM

File: examples/monitor-prom-remotewrite-enable-basic-auth/README.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
+# How to add basic auth to remotewrite example
+
+> **Note:**
+>
+> this example describe how to add basic authorization to tidbMonitor remoteWrite configuration.
+> we use prometheus as remote write receiver,the prometheus managed by [prometheus-operator](https://github.com/coreos/prometheus-operator).
```
Comment (csuzhangxc) [inline examples/monitor-prom-remotewrite-enable-basic-auth/README.md:6]:
```suggestion
> we use Prometheus as the remote write receiver, the Prometheus managed by [prometheus-operator](https://github.com/coreos/prometheus-operator).
```

File: examples/monitor-prom-remotewrite-enable-basic-auth/README.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+> we will add authentication to the prometheus remote write receiver by ingress-nginx controller.
```
Comment (csuzhangxc) [inline examples/monitor-prom-remotewrite-enable-basic-auth/README.md:7]: only one line after rendered to HTML.

File: examples/monitor-prom-remotewrite-enable-basic-auth/README.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+Create TiDB:
+```bash
+> kubectl apply -f tidb-cluster.yaml
+```
+
+Create auth secret and TiDBMonitor:
+```bash
+> kubectl apply -f basic-auth.yaml
+> kubectl apply -f tidb-monitor.yaml
```
Comment (csuzhangxc) [inline examples/monitor-prom-remotewrite-enable-basic-auth/README.md:117]: add `-n` for these commands?

File: examples/monitor-prom-remotewrite-enable-basic-auth/README.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+> kubectl delete pods basic-monitor-0 -n <namespace>
+```
+
+### Validate remotewrite prometheus data
+
+Port-Forward remotewrite prometheus svc port 9090 to local port 9090.
+```bash
+> kubectl port-forward svc/basic-prometheus-remote  9090:9090
+```
+The data is writing successfully.
```
Comment (csuzhangxc) [inline examples/monitor-prom-remotewrite-enable-basic-auth/README.md:142]: How about adding any detailed steps?

File: examples/monitor-prom-remotewrite-enable-basic-auth/basic-auth.yaml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+data:
+  username: YWRtaW4=
+  password: YWRtaW4=
+kind: Secret
+metadata:
+  name: basic-auth
+  namespace: default
```
Comment (csuzhangxc) [inline examples/monitor-prom-remotewrite-enable-basic-auth/basic-auth.yaml:8]: How about removing `namespace` from YAML files and using `-n` in commands instead?

File: examples/monitor-prom-remotewrite-enable-basic-auth/ingress-nginx.yaml [hunk 6]
```diff
@@ -0,0 +1,682 @@
...
+  name: ingress-nginx
+---
+apiVersion: v1
+automountServiceAccountToken: true
+kind: ServiceAccount
+metadata:
+  labels:
+    app.kubernetes.io/component: controller
+    app.kubernetes.io/instance: ingress-nginx
+    app.kubernetes.io/managed-by: Helm
```
Comment (csuzhangxc) [inline examples/monitor-prom-remotewrite-enable-basic-auth/ingress-nginx.yaml:17]: How about simplifying these labels?

**Comment by mikechengwei** [comment]
@csuzhangxc Can you help me review it again?Thks.

**Review by csuzhangxc — APPROVED** [review]

File: examples/monitor-prom-remotewrite-enable-basic-auth/ingress-nginx.yaml (hunk 6 above)
Comment (csuzhangxc) [reply examples/monitor-prom-remotewrite-enable-basic-auth/ingress-nginx.yaml:17]: `managed-by: Helm` seems strange 😂

File: examples/monitor-prom-remotewrite-enable-basic-auth/README.md [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+> kubectl apply -f tidb-cluster.yaml -n <namespace>
+```
+
+Create auth secret and TiDBMonitor:
+```bash
+> kubectl apply -f basic-auth.yaml -n <namespace>
+> kubectl apply -f tidb-monitor.yaml -n <namespace>
+```
+
+Add hostAliases to tidbmonitor statefulset:
```
Comment (DanielZhangQD) [inline examples/monitor-prom-remotewrite-enable-basic-auth/README.md:122]: This change will be reverted by the TiDB Operator, right?

Comment (DanielZhangQD) [reply examples/monitor-prom-remotewrite-enable-basic-auth/README.md:122]: Can we just exec into the Pod to change the /etc/hosts temporarily?

Comment (mikechengwei) [reply examples/monitor-prom-remotewrite-enable-basic-auth/README.md:122]: image

we  can't change /etc/hosts temporarily.

**Pull request closed without merging**